tests/compatibility-tests/src/main/resources/serial/serial.groovy

```diff
@@ -24,30 +24,46 @@ import org.apache.activemq.artemis.jms.client.*
 
 file = arg[0]
 method = arg[1]
-System.out.println("File::" + file);
+version = arg[2]
+System.out.println("File::" + file)
 
 
 if (method.equals("write")) {
     cf = new ActiveMQConnectionFactory("tcp://localhost:61616?confirmationWindowSize=1048576&blockOnDurableSend=false");
     queue = new ActiveMQQueue("queue");
     topic = new ActiveMQTopic("topic")
+    if (version.equals("ARTEMIS-SNAPSHOT")) {
+        destination = new ActiveMQDestination("address", "name", ActiveMQDestination.TYPE.DESTINATION, null)
+    } else if (version.equals("ARTEMIS-155")) {
+        destination = new ActiveMQDestination("address", "name", false, true, null)
+    }
 
     ObjectOutputStream objectOutputStream = new ObjectOutputStream(new FileOutputStream(file));
     objectOutputStream.writeObject(cf);
     objectOutputStream.writeObject(queue)
     objectOutputStream.writeObject(topic)
+    if (version.startsWith("ARTEMIS")) {
```

Comment:
why check the version here? just keep what it was.

Reply:
I mean... why not simply doing objectOutputStream.writeObject(destination)


version will always start with ARTEMIS

Reply:
The link the between the script and the caller seems kind of fragile so I put those checks in there in case the caller was ever changed it wouldn't blow up immediately.